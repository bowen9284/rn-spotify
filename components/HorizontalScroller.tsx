import React from 'react';
import { StyleSheet, View } from 'react-native';
import { ScrollView } from 'react-native-gesture-handler';
import { PrimaryText } from './inputs/PrimaryText';

type Props = {
  title: string;
  items: React.ReactNode;
};

const HorizontalScroller: React.FC<Props> = ({ title, items }) => {

  return (
    <View style={styles.container}>
      <PrimaryText style={styles.headerText}>{title}</PrimaryText>
      <ScrollView horizontal={true}>{items}</ScrollView>
    </View>
  );
};

export default HorizontalScroller;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    height: 225,
  },
  headerText: {
    fontSize: 22,
    fontWeight: 'bold',
    paddingVertical: 15,
  },
});
