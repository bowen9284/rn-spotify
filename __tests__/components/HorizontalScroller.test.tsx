import React from 'react';
import renderer from 'react-test-renderer';
import HorizontalScroller from '../../components/HorizontalScroller';

const props = {
  title: 'Title',
  items: [],
};

test('renders correctly', () => {
  const tree = renderer.create(<HorizontalScroller {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
