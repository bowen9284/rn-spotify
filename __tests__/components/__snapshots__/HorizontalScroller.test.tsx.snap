// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 225,
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(28, 28, 30)",
        "fontSize": 22,
        "fontWeight": "bold",
        "paddingVertical": 15,
      }
    }
  >
    Title
  </Text>
  <RCTScrollView
    collapsable={false}
    horizontal={true}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
  >
    <View />
  </RCTScrollView>
</View>
`;
